### ISSUE TITLE
vision: calling http://169.254.169.254/computeMetadata when running locally

### ISSUE THREAD

**Issue by namco1992** [googleapis/google-cloud-go#2652]
**Client**

vision

**Environment**

e.g. OSX 10.14

**Go Environment**

$ go version
`go version go1.13.9 darwin/amd64`
$ go env
```
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/gongmengnan/Library/Caches/go-build"
GOENV="/Users/gongmengnan/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/gongmengnan/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/Cellar/go@1.13/1.13.9/libexec"
GOSUMDB="off"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go@1.13/1.13.9/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/kk/3k15kpss66b5nh1xhq36cbhc0000gp/T/go-build325961864=/tmp/go-build -gno-record-gcc-switches -fno-common"
```
**Code**

Google cloud documentation example code (https://cloud.google.com/vision/docs/handwriting)
```go

// detectDocumentText gets the full document text from the Vision API for an image at the given file path.
func detectDocumentText(w io.Writer, file string) error {
        ctx := context.Background()

        client, err := vision.NewImageAnnotatorClient(ctx)
        if err != nil {
                return err
        }

        f, err := os.Open(file)
        if err != nil {
                return err
        }
        defer f.Close()

        image, err := vision.NewImageFromReader(f)
        if err != nil {
                return err
        }
        annotation, err := client.DetectDocumentText(ctx, image, nil)
        if err != nil {
                return err
        }

        if annotation == nil {
                fmt.Fprintln(w, "No text found.")
        } else {
                fmt.Fprintln(w, "Document Text:")
                fmt.Fprintf(w, "%q\n", annotation.Text)

                fmt.Fprintln(w, "Pages:")
                for _, page := range annotation.Pages {
                        fmt.Fprintf(w, "\tConfidence: %f, Width: %d, Height: %d\n", page.Confidence, page.Width, page.Height)
                        fmt.Fprintln(w, "\tBlocks:")
                        for _, block := range page.Blocks {
                                fmt.Fprintf(w, "\t\tConfidence: %f, Block type: %v\n", block.Confidence, block.BlockType)
                                fmt.Fprintln(w, "\t\tParagraphs:")
                                for _, paragraph := range block.Paragraphs {
                                        fmt.Fprintf(w, "\t\t\tConfidence: %f", paragraph.Confidence)
                                        fmt.Fprintln(w, "\t\t\tWords:")
                                        for _, word := range paragraph.Words {
                                                symbols := make([]string, len(word.Symbols))
                                                for i, s := range word.Symbols {
                                                        symbols[i] = s.Text
                                                }
                                                wordText := strings.Join(symbols, "")
                                                fmt.Fprintf(w, "\t\t\t\tConfidence: %f, Symbols: %s\n", word.Confidence, wordText)
                                        }
                                }
                        }
                }
        }

        return nil
}
```

**Expected behavior**

The call to deletect document texts should be success.

**Actual behavior**

```bash
rpc error: code = Unauthenticated desc = transport: Get http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/token?scopes=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcloud-platform%2Chttps%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcloud-vision: dial tcp 169.254.169.254:80: connect: host is down
```

**Additional context**
I understood that it called `169.254.169.254` which is only available when running in GCP. However, I was running it locally so it's unreachable. I'm not sure if it's the expected behaviour but I was expecting to run it locally.

**Comment by codyoss** [comment]
Hey, 
It does look light something is going wrong here. It seems the code might think you are running on GCP.
Could you please provide me with what version of `cloud.google.com/go` and `google.golang.org/api` you are pulling in?

**Comment by codyoss** [comment]
Do you by change have the following environment variable set: `GCE_METADATA_HOST`?
[This is the code](https://github.com/googleapis/google-cloud-go/blob/master/compute/metadata/metadata.go#L117) that we use to check if you are running on GCE.

**Comment by namco1992** [comment]
hello @codyoss ,

I found the issue after some debugging. The reason that it thought it's running on GCP is that the VPN I'm using pushed the kube DNS server IP to my local machine, which makes the `metadata.google.internal` resolvable. However, the IP `169.254.169.254` it resolves to is not available outside of the GCP. After I disconnected the VPN it works as expected.

This is not a bug but I feel it's a bit confusing and the API client should have a more deterministic way to check if it's running on GCP.

Thank you for the help and I will close this issue.
